src/codegen/wasm.zig

```diff
@@ -455,64 +465,25 @@ pub const Context = struct {
         try self.emitWValue(lhs);
         try self.emitWValue(rhs);
 
-        const opcode_maybe: ?wasm.Opcode = switch (op) {
-            .lt => @as(?wasm.Opcode, switch (ty) {
-                .i32 => .i32_lt_s,
-                .u32 => .i32_lt_u,
-                .i64 => .i64_lt_s,
-                .u64 => .i64_lt_u,
-                .f32 => .f32_lt,
-                .f64 => .f64_lt,
-                else => null,
-            }),
-            .lte => @as(?wasm.Opcode, switch (ty) {
-                .i32 => .i32_le_s,
-                .u32 => .i32_le_u,
-                .i64 => .i64_le_s,
-                .u64 => .i64_le_u,
-                .f32 => .f32_le,
-                .f64 => .f64_le,
-                else => null,
-            }),
-            .eq => @as(?wasm.Opcode, switch (ty) {
-                .i32, .u32 => .i32_eq,
-                .i64, .u64 => .i64_eq,
-                .f32 => .f32_eq,
-                .f64 => .f64_eq,
-                else => null,
-            }),
-            .gte => @as(?wasm.Opcode, switch (ty) {
-                .i32 => .i32_ge_s,
-                .u32 => .i32_ge_u,
-                .i64 => .i64_ge_s,
-                .u64 => .i64_ge_u,
-                .f32 => .f32_ge,
-                .f64 => .f64_ge,
-                else => null,
-            }),
-            .gt => @as(?wasm.Opcode, switch (ty) {
-                .i32 => .i32_gt_s,
-                .u32 => .i32_gt_u,
-                .i64 => .i64_gt_s,
-                .u64 => .i64_gt_u,
-                .f32 => .f32_gt,
-                .f64 => .f64_gt,
-                else => null,
-            }),
-            .neq => @as(?wasm.Opcode, switch (ty) {
-                .i32, .u32 => .i32_ne,
-                .i64, .u64 => .i64_ne,
-                .f32 => .f32_ne,
-                .f64 => .f64_ne,
-                else => null,
-            }),
-        };
-
-        const opcode = opcode_maybe orelse
-            return self.fail(inst.base.src, "TODO - Wasm genCmp for type '{s}' and operator '{s}'", .{ ty, @tagName(op) });
-
-        try self.code.append(wasm.opcode(opcode));
-        return WValue{ .code_offset = offset };
+        switch (ty) {
+            .i32, .u32, .i64, .u64, .f32, .f64 => {
```

Comment:
Same as my first comment. Have `typeToValtype` handle the type so we can remove this switch.

Reply:
Done.